packages/go_router/lib/src/route_data.dart

```diff
@@ -390,6 +390,28 @@ class TypedGoRoute<T extends GoRouteData> extends TypedRoute<T> {
   final List<TypedRoute<RouteData>> routes;
 }
 
+/// A superclass for each typed go route descendant
+@Target(<TargetKind>{TargetKind.library, TargetKind.classType})
+class TypedRelativeGoRoute<T extends GoRouteData> extends TypedRoute<T> {
```

Comment:
I can't see the difference between this  and `TypedGoRoute` it seems to me that if user uses go_router_builder they are not using url string to navigate so they don't really need relative routing?

Reply:
Well they won't be using './$path' directly, but they'll use `route.goRelative()`, which uses that under the hood.

Not sure if I got your question right thou..

Reply:
I meant they could define a TypedGoRoute directly and do GoRouteData.go, why would they want to define a TypedRelativeRoute ?

Reply:
They can't define GoRoute in multiple places. Take the example in the description, the relativeRoute was declared in a variable and passed in 2 routes. Had that been normal GoRoute, it would've produced 2 extensions with the same name, of which locations are different, and results in build time error

Reply:
This seems to be two different feature right?

1. is to be able to do `go('./relative')`
2. is to be able to reuse the same RouteData with same relative path in different sub route.

and 2 needs to build on top of 1.
If so I think we should separate the 2 into a new pr to make review simpler.

Reply:
oh looks like you already did https://github.com/flutter/packages/pull/6825.

Reply:
okay... so do I need to adjust anything?

Reply:
okay... so do I need to change anything else?

Reply:
I would suggest removing the TypeRelativeRoute from https://github.com/flutter/packages/pull/6825 and do all the TypeRelativeRoute thing in a separate pr

Reply:
TypedRelativeRoute needs to be landed in order for this PR to work. The reason this PR is working right now is because of the `dependency_overrides`, which I meant to remove after the #6825 is merged. Can you suggest how do I remove the TypedRelativeRoute? Do I create a PR with only `TypedRelativeRoute` declaration, wait for it to get merged and then continue with this PR?

Imo it's feels weird to me that an annotation of a builder package belongs to the main package and not in the builder package, but maybe there is a technical reason for that that I don't know

Reply:
No I meant the TypedRelativeRoute should be separate from https://github.com/flutter/packages/pull/6825 since they are not related. and then we can review and merge https://github.com/flutter/packages/pull/6825 first before we review TypedRelativeRoute separately.